metrics-healthchecks/src/main/java/com/codahale/metrics/health/HealthCheckRegistry.java

```diff
@@ -105,7 +105,9 @@ public void register(String name, HealthCheck healthCheck) {
                 healthChecks.put(name, registered);
             }
         }
-        if (registered != null) {
+        if (registered == null) {
+            throw new IllegalArgumentException("A health check named " + name + " already exists");
+        } else {
```

Comment:
It seems to me that it would be to move throwing exception near to the `containsKey` check. It looks more logical to me (we check that the health check doesn't exist, if not throw an exception). 
```java
if (!healthChecks.containsKey(name)) {
  ....
} else {
  throw new IllegalArgumentException("A health check named " + name + " already exists");
}
```

Also this allows to remove the not null check before invoking the listeners, since the compiler will guarantee that `registered` will be always not null.